### PR TITLE
'solution'

### PR DESCRIPTION
**Pull request by IvanDeiko**

### CODE REVIEW

File: src/index.html [hunk 1]
```diff
@@ -0,0 +1,12 @@
...
+<head>
+  <meta charset="UTF-8">
+  <meta name="viewport" content="width=device-width, initial-scale=1.0">
+  <meta http-equiv="X-UA-Compatible" content="ie=edge">
+  <title>Document</title>
+</head>
+<body>
+  <script src="main.js"></script>
+</body>
+</html>
```
Comment (gestap) [inline src/index.html]: the document must end with one empty line

File: src/main.js [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+
+}
+
+let newList = new Numberdrome();
+
+
+
+
+newList.addNumber(32);
+console.log(newList.max())
```
Comment (gestap) [inline src/main.js]: the document must end with one empty line

File: src/main.js [hunk 3]
```diff
@@ -0,0 +1,56 @@
+class Numberdrome {
```
Comment (ost-ptk) [inline src/main.js]: use strict mode

File: src/main.js [hunk 4]
```diff
@@ -0,0 +1,56 @@
...
+  }
+
+  product() {
+    return this.listOfNumbers.reduce((sum, current) => {
+      return sum * current;
+    }, 1);
+  }
+  
+  _funcForMaxAndMin(maxOrMin) {
+    try {
```
Comment (gestap) [inline src/main.js]: no need to use `try...catch`

File: src/main.js [hunk 5]
```diff
@@ -0,0 +1,56 @@
...
+      if (!this.listOfNumbers.length) {
+        throw new Error("List of number doesn't have numbers");
+      };
+
+      return maxOrMin === 'max' ? Math.max(...this.listOfNumbers) :
+                                  Math.min(...this.listOfNumbers);
+    } catch(e) {
+      return e.message;
```
Comment (gestap) [inline src/main.js]: you need `throw an error` instead of return message

File: src/main.js (hunk 2 above)
Comment (gestap) [inline src/main.js]: remove unnecessary code

File: src/main.js [hunk 6]
```diff
@@ -0,0 +1,56 @@
...
+  }
+
+  max() {
+    return this._funcForMaxAndMin('max');
+  }
+
+}
+
+let newList = new Numberdrome();
+
```
Comment (gestap) [inline src/main.js]: remove empty line

**Review by ost-ptk — CHANGES_REQUESTED** [review]

File: src/main.js [hunk 7]
```diff
@@ -0,0 +1,49 @@
...
+      return sum * current;
+    }, 1);
+  }
+  
+  _funcForMaxAndMin(maxOrMin) {
+    
+      if (!this.listOfNumbers.length) {
+        throw new Error("List of number doesn't have numbers");
+      } else {
+        return maxOrMin === 'max' ? Math.max(...this.listOfNumbers) :
```
Comment (gestap) [inline src/main.js]:
```suggestion
        return maxOrMin === 'max' 
            ? Math.max(...this.listOfNumbers) 
            : Math.min(...this.listOfNumbers);
```

File: src/main.js [hunk 8]
```diff
@@ -0,0 +1,49 @@
...
+  }
+
+  product() {
+    return this.listOfNumbers.reduce((sum, current) => {
+      return sum * current;
+    }, 1);
+  }
+  
+  _funcForMaxAndMin(maxOrMin) {
+
```
Comment (gestap) [inline src/main.js]: remove empty line

File: src/main.js [hunk 9]
```diff
@@ -0,0 +1,49 @@
...
+  
+  _funcForMaxAndMin(maxOrMin) {
+    
+      if (!this.listOfNumbers.length) {
+        throw new Error("List of number doesn't have numbers");
+      } else {
+        return maxOrMin === 'max' ? Math.max(...this.listOfNumbers) :
+                                    Math.min(...this.listOfNumbers);
+      }
+
```
Comment (gestap) [inline src/main.js]: remove empty line

**Review by ost-ptk — CHANGES_REQUESTED** [review]

**Review by ost-ptk — APPROVED** [review]

**Pull request closed without merging**